### PR TITLE
feat(app-platform): add modal for permissions

### PR DESCRIPTION
**Pull request by MeredithAnya**
This modal is triggered when clicking **Install** on the organization integrations page for any sentry apps. It shows the scopes (in hopefully a reasonably digest-able way) that the app is requesting.

The resources (`org`, `project`, etc.) are shown under their 'highest level' permission. So under the hood '**Admin** access to member'  means they have `member:read`, `member:write` and `member:admin`.

If an app has a `redirect_url` the text at the bottom will show up letting the user know that they will be redirected, otherwise only the buttons will be there. 

>screen shot 2018-12-10 at 9 47 00 am

In Sentry, `project:releases` is a weird scope because either you have it or you don't, there isn't `releases:<read, write, admin>`. So we are just going to add the Releases resource to **Admin**

(cc @benvinegar )

### CODE REVIEW

File: src/sentry/static/sentry/app/components/modals/sentryAppPermissionsModal.jsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    return (
+      <React.Fragment>
+        <Header closeButton onHide={closeModal}>
+          {t(`Install ${app.name}`)}
+        </Header>
+        <Body>
+          <Title>
+            {'Install on your '}
+            <strong>{orgId}</strong>
+            {' organization with the following permissions:'}
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/modals/sentryAppPermissionsModal.jsx]: Should these have `t()`?

File: src/sentry/static/sentry/app/components/modals/sentryAppPermissionsModal.jsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+          </Title>
+          <Panel>{this.renderPermissions()}</Panel>
+        </Body>
+        <div className="modal-footer">
+          {!app.redirectUrl && (
+            <RedirectionInfo>
+              {`After installation you'll be redirected to the ${app.name} service to finish setup.`}
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/modals/sentryAppPermissionsModal.jsx]: t()?

File: src/sentry/static/sentry/app/components/modals/sentryAppPermissionsModal.jsx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        </div>
+      </React.Fragment>
+    );
+  }
+}
+
+export default SentryAppPermissionsModal;
+
+const StyledInlineSvg = styled(InlineSvg)`
+  margin-left: 8px;
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/modals/sentryAppPermissionsModal.jsx]: Should this be using `space()`?

**Comment by adhiraj** [comment]
Do we need to expose the fact that releases doesn't have a read/write/admin internally (so just say "Admin access to Releases" to be consistent)?

File: src/sentry/static/sentry/app/views/organizationIntegrations/sentryAppInstallations.jsx [hunk 4]
```diff
@@ -45,54 +52,41 @@ export default class SentryAppInstallations extends AsyncView {
...
-      success,
-      error,
-    };
+    uninstallSentryApp(api, install).then(
+      () => {},
+      () => {
+        this.setState({origInstalls});
+        addErrorMessage(t(`Unable to uninstall ${install.app.name}`));
+      }
+    );
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/organizationIntegrations/sentryAppInstallations.jsx]: If you only need to set an error handler you can use `promise.catch()`. This works as well though.

File: src/sentry/static/sentry/app/views/organizationIntegrations/sentryAppInstallations.jsx [hunk 5]
```diff
@@ -115,3 +109,6 @@ export default class SentryAppInstallations extends AsyncView {
     );
   }
 }
+
+export default withApi(SentryAppInstallations);
+export {SentryAppInstallations};
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/organizationIntegrations/sentryAppInstallations.jsx]: Should the named and default export be different?

File: tests/js/spec/views/settings/organizationIntegrations/sentryAppInstallations.spec.jsx [hunk 6]
```diff
@@ -93,20 +128,22 @@ describe('Sentry App Installations', function() {
         );
 
         wrapper.find('[icon="icon-circle-add"]').simulate('click');
-
+        wrapper.instance().install(sentryApp);
+        await tick;
```
Comment (markstory) [inline tests/js/spec/views/settings/organizationIntegrations/sentryAppInstallations.spec.jsx]: Do you need to call tick()?

**Review by markstory — APPROVED** [review]
Looks good to me.

**Review by mnoble — COMMENTED** [review]
Looks really good overall. Some small suggestions.

**Pull request merged**